src/main/java/com/github/brainlag/nsq/lookup/DefaultNSQLookup.java

```diff
@@ -26,16 +27,29 @@ public void addLookupAddress(String addr, int port) {
         Set<ServerAddress> addresses = Sets.newHashSet();
 
         for (String addr : this.addresses) {
-            ObjectMapper mapper = new ObjectMapper();
-
-            JsonNode jsonNode = mapper.readTree(new URL(addr + "/lookup?topic=" + topic));
-            JsonNode producers = jsonNode.get("data").get("producers");
-            for (JsonNode node : producers) {
-                String host = node.get("broadcast_address").asText();
-                ServerAddress address = new ServerAddress(host, node.get("tcp_port").asInt());
-                addresses.add(address);
+            try {
+                ObjectMapper mapper = new ObjectMapper();
+
+                JsonNode jsonNode = mapper.readTree(new URL(addr + "/lookup?topic=" + topic));
+                LogManager.getLogger(this).warn("Server connection information: " + jsonNode.toString());
+                JsonNode producers = jsonNode.get("data").get("producers");
+                for (JsonNode node : producers) {
+                    String host = node.get("broadcast_address").asText();
+                    ServerAddress address = new ServerAddress(host, node.get("tcp_port").asInt());
+                    addresses.add(address);
+                }
+            } catch (IOException e) {
+                LogManager.getLogger(this).warn("Unable to connect to address " + addr);
```

Comment:
I added this here since during my testing of handling multiple nsqd lookup servers, if none of the servers were reachable (either wrong IP, etc), it was failing silently, which for our application wasn't acceptable since we would want to be notified if our application couldn't connect to a nsqd lookup server.